employee-rostering-frontend/src/store/contract/operations.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ThunkCommandFactory } from '../types';
+import * as actions from './actions';
+import Contract from 'domain/Contract';
+import { AddContractAction, RemoveContractAction, UpdateContractAction, RefreshContractListAction } from './types';
+import { employeeOperations } from 'store/employee';
+
+export const addContract: ThunkCommandFactory<Contract, AddContractAction> = contract =>
+  (dispatch, state, client) => {
+    let tenantId = contract.tenantId;
+    return client.post<Contract>(`/tenant/${tenantId}/contract/add`, contract).then(newContract => {
+      dispatch(actions.addContract(newContract))
```

Comment:
Actually, you can simply return the action here instead of dispatching it. You'd need to create an new type for the factory method (something like `ThunkActionFactory`) but that's expected. ThunkAction is a function that returns `Promise<SomeAction>`. I use ThunkCommand (which is void) in VRP because sending a message over WebSocket has no direct response, so there's no response data to create an action from.

Reply:
I think we can take a look at this later after this PR.